src/data/nat/factorization.lean

```diff
@@ -93,6 +93,20 @@ end
   factorization (p^k) = single p k :=
 by simp [factorization_pow, hp.factorization]
 
+/-- For any `p : ℕ` and any function `g : α → ℕ` that's non-zero on `S : finset α`,
+the power of `p` in `S.prod g` equals the sum over `x ∈ S` of the powers of `p` in `g x`.
+Generalises `factorization_mul`, which is the special case where `S.card = 2` and `g = id` -/
+lemma factorization_prod {α : Type*} {S : finset α} {g : α → ℕ} (hS: ∀ x ∈ S, g x ≠ 0) :
```

Comment:
```suggestion
Generalises `factorization_mul`, which is the special case where `S.card = 2` and `g = id`. -/
lemma factorization_prod {α : Type*} {S : finset α} {g : α → ℕ} (hS : ∀ x ∈ S, g x ≠ 0) :
```
It looks like bors ignored Chris's comment for some reason (or possibly the additional commit to this branch took it off the queue and bors forgot to say "Canceled"), so I figured we might as well squeeze this style nitpick in.

bors d+